jupyter_client/event_schemas/kernel_manager/v1.yaml

```diff
@@ -0,0 +1,36 @@
+"$id": https://events.jupyter.org/jupyter_client/kernel_manager/v1
+version: 1
+title: Kernel Manager Events
+description: |
+  Record actions on kernels by the KernelManager.
+type: object
+required:
+  - kernel_id
+  - action
+properties:
+  kernel_id:
+    oneOf:
+      - type: string
+      - type: "null"
```

Comment:
This looks good @Zsailer - just have one question.

Is this in _anticipation_ of supporting other actions, because all of the currently defined actions fire with a non-None `kernel_id`?  If we retained that approach, we could then apply a format or pattern, allowing only UUID formats (and remove the `oneOf` and `null`).  This might allow event consumers to have an easier experience as well.

Reply:
> because all of the currently defined actions fire with a non-None kernel_id?

I don't think `start_kernel` or `pre_start_kernel` methods strictly require a kernel_id to start, right? [This logic](https://github.com/jupyter/jupyter_client/blob/ff85055a3b0ab198e3b1250185628b5c4929615e/jupyter_client/manager.py#L342) seems to suggest that if `None` is provided, it creates a uuid on the fly.

Reply:
That is correct (on both counts; non-required kernel_id and auto-create logic).  The provisioners should have access to the kernel-id from the get-go so they can do indexing, etc. as necessary.